src/Overlay.js

```diff
@@ -42,56 +35,63 @@ class Overlay extends React.Component {
       , animation: Transition
       , ...props } = this.props;
 
-    let child = null;
-
-    if ( Transition === true ){
+    if (Transition === true) {
       Transition = Fade;
     }
 
-    if (props.show || (Transition && this.state.exiting && !this.state.exited)) {
+    // Don't un-render the overlay while it's transitioning out.
+    const mountOverlay = props.show || (Transition && !this.state.exited);
+
+    if (!mountOverlay) {
+      // Don't bother showing anything if we don't have to.
+      return null;
+    }
 
-      child = children;
+    let child = children;
 
-      // Position the child before the animation to avoid `null` DOM nodes
+    if (Transition) {
+      // This animates the child by injecting props, so it must be inner-most.
       child = (
-        <Position {...{ container, containerPadding, target, placement }}>
-          { child }
-        </Position>
+        <Transition
+          in={props.show}
+          transitionAppear
```

Comment:
I was more or less trying to copy the meaning from [CSSTransitionGroup](http://facebook.github.io/react/docs/animation.html#animate-initial-mounting) which, now that I think about it, only apply for entering, since state is determined by whether children exist there. Which is to say matches your interpretation.

I was thinking originally though that since the `in` prop determines initial state, it could also apply to the inverse, i.e:  "if the component is not `in` on mounting, mount it in the entered state, but transition to exiting immediately after the initial render". That was the original behavior. That being said its a totally valid to question whether that makes any sense. Would you ever want to animate out on mount? I went for just allowing it for flexibility, but I can't personally think of an actual use case...